recipes/Code_Gen_from_Tests/Code_Gen_from_Tests.ipynb

```diff
@@ -0,0 +1,496 @@
+    "\n",
+    "The notebook depends on the Granite [utils](https://github.com/ibm-granite-community/utils) package for integration with LLMs using the [Langchain](https://www.langchain.com/) framework.\n",
+    "\n",
+    "> **TIP:** See the [Getting Started with Replicate](https://github.com/ibm-granite-community/granite-kitchen/blob/main/recipes/Getting_Started/Getting_Started_with_Replicate.ipynb) notebook in the [granite-kitchen](https://github.com/ibm-granite-community/granite-kitchen) repo for more information about using Replicate."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Install the required Langchain and Replicate packages\n",
+    "\n",
+    "Include a granite-community package with some simple utility functions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "!pip install git+https://github.com/ibm-granite-community/utils \\\n",
+    "    \"langchain_community<0.3.0\" \\\n",
+    "    replicate"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ibm_granite_community.notebook_utils import set_env_var, get_env_var"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate the Code from a Hypothesis _Property-Based_ Test Suite\n",
+    "\n",
+    "We will use a testing library called [Hypothesis](https://hypothesis.readthedocs.io/en/latest/) that helps you think about the _properties_ the unit under test must satisfy. We will define some Hypothesis tests in a string below, then save it to a file so we can execute the tests. They cover a `Rational` class (for rational numbers) that doesn't exist yet. Finally, we will use Granite to generate an implementation of `Rational` that hopefully allows the tests to pass.  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The next two cells install `hypothesis` and load the first set of tests for `Rational` from a file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "!pip install 'hypothesis[cli]'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we define our Hypothesis tests as a string and then save the string to a file so we can run the tests later. (Note that we have to use a hack to nest `\"\"\"` strings.)\n",
+    "\n",
+    "> **Note:** This test file is adapted from [this GitHub project](https://github.com/deanwampler/tdd-hypothesis-example)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tests = \"\"\"\n",
```

Comment:
```suggestion
    "tests = '''\n",
```

Use triple single quotes so you don't have to escape the triple double quotes.

Reply:
I didn't escape them. This is done by Jupyter to write the notebook's JSON file. The escaping is not in the notebook cell itself.

Reply:
Inside the string you had to `\"""` the embedded triple double quotes. This is not needed if you use triple single quotes at the outer level.

Reply:
True. I fixed that. Thanks.